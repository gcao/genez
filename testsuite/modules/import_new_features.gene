#!/usr/bin/env gene run

# Test new import features from packages_and_modules.md

# Test 1: New => alias syntax (already works)
(import "math" => m)
(import "utils/helpers" => h)

# Test 2: Import with array and rename
# (import "utils/helpers" [helper1 => h1 helper2])

# Test 3: Import all (not recommended)
# (import "utils/helpers" *)

# Test 4: Well-known modules (future feature)
# (import "gene/core")
# (import "genex/http")

# Test 5: Private function syntax (future feature)
# (fn- private-helper [x] (* x 2))

# Test 6: Module exports (future feature)
# ^module-exports [add subtract multiply]

(print "New import features test completed")