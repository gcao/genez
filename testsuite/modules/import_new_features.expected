New import features test completed